<launch>
  <!-- Launchfile Arguments (optional)-->
  <arg name="robot_name" default="WAM"/>

  <!-- Publish TF Frames for all the joints -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher"/>

  <!-- Parameters for WAM RTT stub -->
  <!-- Robot Model & Kinematics-->
  <!--param name="robot_description" command="$(find xacro)/xacro.py '$(find bard_urdf)/robots/wam_7dof_$(arg robot_name).urdf.xacro' robot_name:=$(arg robot_name)"/-->
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find bard_urdf)/robots/wam_7dof_$(arg robot_name).urdf.xacro'"/>
  <param name="root_link" value="$(arg robot_name)/FixedLink"/>
  <param name="tip_link" value="$(arg robot_name)/LowerWristYawLink"/>

  <!-- Component Configuration -->
  <rosparam param="initial_positions">{Element0: 0.0, Element1: rad(-pi/2), Element2: 0.0, Element3: rad(pi), Element4: 0.0, Element5: 0.0, Element6: 0.0}</rosparam>
  <param name="joint_state_throttle_period" value="0.01"/>

  <!-- Simulation Options -->
  <param name="coriolis_enabled" value="true"/>
  <param name="gravity_enabled" value="true"/>
  <rosparam param="gravity">{Element0: -9.8, Element1: 0.0, Element2: 0.0}</rosparam>
  <rosparam param="damping">{Element0: 0.3, Element1: 0.3, Element2: 0.3, Element3: 0.3, Element4: 0.1, Element5: 0.1, Element6: 0.1}</rosparam>
</launch>
